pkg/operator/controller/garden/garden/components.go

```diff
@@ -651,7 +651,33 @@ func (r *Reconciler) newKubeAPIServer(
 	)
 }
 
-func defaultAPIServerAutoscalingConfig(garden *operatorv1alpha1.Garden) apiserver.AutoscalingConfig {
+func gardenerAPIServerAutoscalingConfig(garden *operatorv1alpha1.Garden) apiserver.AutoscalingConfig {
+	replicas := int32(2)
+	if helper.HighAvailabilityEnabled(garden) {
+		replicas = 3
+	}
+
+	var minAllowed corev1.ResourceList
+	if kubeAPIServer := garden.Spec.VirtualCluster.Kubernetes.KubeAPIServer; kubeAPIServer != nil && kubeAPIServer.Autoscaling != nil {
+		minAllowed = kubeAPIServer.Autoscaling.MinAllowed
+	}
```

Comment:
This is strange. Why do we take KAPI configuration for GAPI? I see this was the case before as well, but I guess it's wrong/a mistake?

Reply:
This is not just for autoscaling. There are values for apiserver in general ([ref](https://github.com/gardener/gardener/blob/085ac49989fdce2a7f93ddad5bbc9b9c64e5ef11/pkg/component/apiserver/types.go#L30)) which are embedded into kube-apiserver ([ref](https://github.com/gardener/gardener/blob/085ac49989fdce2a7f93ddad5bbc9b9c64e5ef11/pkg/component/kubernetes/apiserver/apiserver.go#L84)) and gardener-apiserver ([ref](https://github.com/gardener/gardener/blob/085ac49989fdce2a7f93ddad5bbc9b9c64e5ef11/pkg/component/gardener/apiserver/apiserver.go#L56)).

Reply:
I found that `gardener-apiserver` is using only 2 of 6 fields of the generic `AutoscalingConfig` values structure.
Thus, I created separate config structures for gardener and kube-apiservers to avoid confusion about the supported settings.